src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -113,125 +137,269 @@ internal abstract class AbstractAddParameterCodeFixProvider<
+                        context.Diagnostics);
+                }
+                else
+                {
+                    context.RegisterCodeFix(
+                        new MyCodeAction(title,
+                            c => FixAsync(context.Document, methodToUpdate, argumentToInsert, arguments, fixAllReferences: false, c)),
+                        context.Diagnostics);
+                }
+            }
+        }
 
-                var title = string.Format(FeaturesResources.Add_parameter_to_0, signature);
+        /// <summary>
+        /// Checks if there are indications that there might be more than one declarations that need to be fixed.
+        /// The check does not look-up if there are other declarations (this is done later in the CodeAction).
+        /// </summary>
+        private bool HasCascadingDeclarations(IMethodSymbol method)
+        {
+            // Don't cascade constructors
+            if (method.IsConstructor())
+            {
+                return false;
+            }
+
+            // Virtual methods of all kinds might have overrides somewhere else that need to be fixed.
+            if (method.IsVirtual || method.IsOverride || method.IsAbstract)
+            {
+                return true;
+            }
 
-                context.RegisterCodeFix(
-                    new MyCodeAction(title, c => FixAsync(document, constructor, argumentToAdd, arguments, c)),
-                    context.Diagnostics);
+            // If interfaces are involved we will fix those too
+            // Explicit interface implementations are easy
+            if (method.ExplicitInterfaceImplementations.Length > 0)
+            {
+                return true;
             }
+
+            // For implicit interface implementations lets check if the characteristic of the method
+            // allows it to implicit implement an interface member.
+            if (method.DeclaredAccessibility == Accessibility.Private || method.DeclaredAccessibility == Accessibility.NotApplicable)
+            {
+                return false;
+            }
+
+            if (method.IsStatic)
+            {
+                return false;
+            }
+
+            // Now check if the method does implement an interface member
+            if (method.ExplicitOrImplicitInterfaceImplementations().Length > 0)
+            {
+                return true;
+            }
+
+            return false;
         }
 
-        private int NonParamsParameterCount(IMethodSymbol method)
-            => method.IsParams() ? method.Parameters.Length - 1 : method.Parameters.Length;
+        private static string GetCodeFixTitle(string resourceString, IMethodSymbol methodToUpdate, IEnumerable<string> parameters)
+        {
+            var methodPrefix = methodToUpdate.IsConstructor()
+                ? ""
+                : $"{methodToUpdate.ContainingType.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat)}.";
```

Comment:
why not just get the display string for the method?  i think there are flags saying: i do not want parameters included.